Source/Tests/LinuxSyscalls/SyscallsSMCTracking.cpp

```diff
@@ -0,0 +1,326 @@
+/*
+$info$
+category: LinuxSyscalls ~ Linux syscall emulation, marshaling and passthrough
+tags: LinuxSyscalls|common
+desc: SMC/MMan Tracking
+$end_info$
+*/
+
+#include <filesystem>
+#include <sys/shm.h>
+#include <sys/mman.h>
+
+#include "Tests/LinuxSyscalls/Syscalls.h"
+
+#include <FEXHeaderUtils/TypeDefines.h>
+#include <FEXCore/Debug/InternalThreadState.h>
+#include <FEXCore/Utils/LogManager.h>
+#include <FEXCore/Utils/MathUtils.h>
+
+namespace FEX::HLE {
+
+// SMC interactions
+bool SyscallHandler::HandleSegfault(FEXCore::Core::InternalThreadState *Thread, int Signal, void *info, void *ucontext) {
+  auto CTX = Thread->CTX;
+
+  auto FaultAddress = (uintptr_t)((siginfo_t *)info)->si_addr;
```

Comment:
```suggestion
  const auto FaultAddress = (uintptr_t)((siginfo_t *)info)->si_addr;
```

Reply:
You sure do love using const hah

Reply:
For long scopes, yes - much easier to grasp data flow if I know which values don't change. Looks good now :)